File: src/main/java/de/floribe2000/warships_java/direct/clans/Clans.java
package de.floribe2000.warships_java.direct.clans;

import de.floribe2000.warships_java.direct.api.ErrorContainer;
import de.floribe2000.warships_java.direct.api.IApiResponse;
import de.floribe2000.warships_java.direct.api.IRequestAction;
import de.floribe2000.warships_java.direct.api.Meta;
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status;
import lombok.Getter;

import java.util.List;

@Getter
public class Clans implements IApiResponse {

    private Status status = Status.ERROR;

    /**
     * Details about errors in case of a failed request.
     * <p>Field is null if no errors occurred during the request!</p>
     */
    private ErrorContainer error = null;

    private Meta meta = null;

    private List<ClanDetails> data = null;

    @Getter
    public static class ClanDetails {

        private int members_count = 0;

        private long created_at = 0;

        private int clan_id = 0;

        private String tag = null;

        private String name = null;
    }

    @Override
    public String toString() {
        return IRequestAction.GSON.toJson(this);
    }
}

File: src/main/kotlin/de/floribe2000/warships_java/direct/clans/Clans.kt
package de.floribe2000.warships_java.direct.clans

import com.google.gson.annotations.SerializedName
import de.floribe2000.warships_java.direct.api.ErrorContainer
import de.floribe2000.warships_java.direct.api.IApiResponse
import de.floribe2000.warships_java.direct.api.IRequestAction.Companion.GSON
import de.floribe2000.warships_java.direct.api.Meta
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status
import kotlinx.serialization.SerialName

data class Clans(
    override val status: Status = Status.ERROR,

    /**
     * Details about errors in case of a failed request.
     *
     * Field is null if no errors occurred during the request!
     */
    override val error: ErrorContainer? = null,

    override val meta: Meta = Meta(),

    val data: List<ClanDetails> = listOf(),
) : IApiResponse {


    data class ClanDetails(
        @SerializedName("members_count")
        @SerialName("members_count")
        val membersCount: Int,

        @SerializedName("created_at")
        @SerialName("created_at")
        val createdAt: Long,

        @SerializedName("clan_id")
        @SerialName("clan_id")
        val clanId: Long,

        val tag: String,

        val name: String,
    )

    override fun toString(): String {
        return GSON.toJson(this)
    }
}